src/storage/test/QueryBoundTest.cpp

```diff
@@ -272,6 +339,78 @@ TEST(QueryBoundTest,  GenBucketsTest) {
+    mockData(kv.get());
+
+    LOG(INFO) << "Build filter...";
+    auto* tag = new std::string("3001");
+    auto* prop = new std::string("tag_3001_col_0");
+    auto* srcExp = new SourcePropertyExpression(tag, prop);
+    auto* priExp = new PrimaryExpression(20 + 3001L);
+    auto* left = new RelationalExpression(srcExp,
+                                          RelationalExpression::Operator::GE,
+                                          priExp);
+    auto* edgeProp = new std::string("col_0");
+    auto* alias = new std::string("e101");
+    auto* edgeExp = new EdgePropertyExpression(alias, edgeProp);
+    auto* priExp2 = new PrimaryExpression(10007L);
+    auto* right = new RelationalExpression(edgeExp,
+                                           RelationalExpression::Operator::GE,
+                                           priExp2);
+    auto logExp = std::make_unique<LogicalExpression>(left, LogicalExpression::AND, right);
+
+    cpp2::GetNeighborsRequest req;
+    buildRequest(req);
+    req.set_filter(Expression::encode(logExp.get()));
+
+    LOG(INFO) << "Test QueryOutBoundRequest...";
+    auto executor = std::make_unique<folly::CPUThreadPoolExecutor>(3);
+    auto* processor = QueryBoundProcessor::instance(kv.get(),
+                                                    schemaMan.get(),
+                                                    executor.get(),
+                                                    BoundType::OUT_BOUND);
+    auto f = processor->getFuture();
+    processor->process(req);
+    auto resp = std::move(f).get();
+
+    LOG(INFO) << "Check the results...";
+    checkResponse(resp, 10, 12, 10007, 1, true);
+}
+
+TEST(QueryBoundTest, FilterTest_InvalidFilter) {
+    fs::TempDir rootPath("/tmp/QueryBoundTest.XXXXXX");
+    LOG(INFO) << "Prepare meta...";
+    std::unique_ptr<kvstore::KVStore> kv(TestUtils::initKV(rootPath.path()));
+    auto schemaMan = TestUtils::mockSchemaMan();;
+    mockData(kv.get());
+
+    LOG(INFO) << "Build filter...";
+    auto* prop = new std::string("tag_3001_col_0");
+    auto inputExp = std::make_unique<InputPropertyExpression>(prop);
+
+    cpp2::GetNeighborsRequest req;
+    buildRequest(req);
+    req.set_filter(Expression::encode(inputExp.get()));
+
+    LOG(INFO) << "Test QueryOutBoundRequest...";
+    auto executor = std::make_unique<folly::CPUThreadPoolExecutor>(3);
+    auto* processor = QueryBoundProcessor::instance(kv.get(),
+                                                    schemaMan.get(),
+                                                    executor.get(),
+                                                    BoundType::OUT_BOUND);
+    auto f = processor->getFuture();
+    processor->process(req);
+    auto resp = std::move(f).get();
+
+    LOG(INFO) << "Check the results...";
+    EXPECT_EQ(3, resp.result.failed_codes.size());
+    EXPECT_TRUE(nebula::storage::cpp2::ErrorCode::E_INVALID_FILTER
+                    == resp.result.failed_codes[0].code);
+}
```

Comment:
I think only (some) relational & logical expressions are test, what about others?
Besides, the data writen in and filter out are hard to check. I hope you check the result carefully.

Reply:
The UTs here is to test edgePropFilter and tagPropFilter.  Only the two basic filters is handled in storage layer.  All other filters have been tested in the UTs of Expression module.